test/micro/case_test.rb

```diff
@@ -150,4 +150,106 @@ def test_inspect
       Multiply.inspect
     )
   end
+
+  class Bomb < Micro::Case
+    attributes :defused
+
+    def call!
+      Try do
+        if defused
+          :yay
+        else
+          raise 'Boooom!'
+        end
+      end
+    end
+  end
+
+  def test_try
+    result = Bomb.call(defused: true)
+
+    assert_success_result(result, value: { try: :yay }, type: :ok)
+
+    result = Bomb.call(defused: false)
+
+    assert_failure_result(result, type: :exception)
+    assert result.value[:try].is_a? RuntimeError
+  end
+
+  class BombWithType < Micro::Case
+    attributes :defused
+
+    def call!
+      Try(:defused) do
+        if defused
+          :yay
+        else
+          raise 'Boooom!'
+        end
+      end
+    end
+  end
+  
+  def test_try_with_custom_type
+    result = BombWithType.call(defused: true)
+
+    assert_success_result(result, type: :defused, value: { defused: :yay })
+    
+
+    result = BombWithType.call(defused: false)
+
+    assert_failure_result(result, type: :defused)
+    assert(result.value[:defused].is_a? RuntimeError)
+  end
+
+  class Divide2 < Micro::Case
+    attributes :a, :b
+
+    def call!
+      Try(catch: [ZeroDivisionError, TypeError]) do
+        a / b
+      end
+    end
+  end
+
+  def test_try_with_specified_exception
+    result = Divide2.call(a: 4 , b: 2)
+
+    assert_success_result(result, value: { try: 2 })
+
+    result = Divide2.call(a: 1 , b: 0)
+
+    assert_failure_result(result)
+    assert(result.value[:try].is_a? ZeroDivisionError)
+
+    result = Divide2.call(a: 1 , b: '0')
+
+    assert_failure_result(result)
+    assert(result.value[:try].is_a? TypeError)
+
+
+    assert_raises NoMethodError do
+      Divide2.call(a: '1' , b: 0)
+    end
+  end
+
+  class DivideWithCustomData < Micro::Case
+    attributes :a, :b
+
+    def call!
+      Try(on: { success: { division: :succeeded }, failure: { division: :failed } }) do
```

Comment:
@MatheusRich what do you think if we allow the usage of lambdas to define a dynamic output from `on[:success]/on[:failure]`?

```ruby
Try(:division, on: { success: -> division {{ value: division }} }) do
  a / b
end
```

Reply:
We _could_ do this, but it seems to mix the monad `type` and its `value`. Any particular reason to do that?

Currently, you can do this:
```ruby
division = Try(:division) { a / b }.value

{ value: division }
```
Plus, `Check` doesn't support this as well, and I think these two methods should keep a similar API.

Reply:
Hi @MatheusRich, as we talked via Telegram, your questions made me think about the API and I could figure out that the reason for this is related to the fact of the `Try` returns the block output as the value of its result type (`try` as you suggested). So, the idea of this kind of feature is to enable the user to define another hash instead of the default `{ try: some_value }`.

But, we could resolve all of this if we define a Hash as the block output. e.g.
```ruby
class DivideIntegers < Micro::Case
  attributes :a, :b, default: -> value { value.to_i }

  def call!
    Try do
      { division: a / b }
    end
  end
end
```
What do you think about this?

Reply:
@MatheusRich one more thing that I forgot to say. As `Try` is a word about the tentative of something. I think that its result type could be the same for success or failure. e.g.

```ruby
class DivideIntegers < Micro::Case
  attributes :a, :b, default: -> value { value.to_i }

  def call!
    Try(:division) do
      { value: a / b }
    end
  end
end

# Success(:division, result: { value: 2 })
result = DivideIntegers.call(a: 4, b: '2')
result.success? == true
result.type == :division
result.data == { value: 2 }

# Failure(:division, result: { exception: #<ZeroDivisionError: divided by 0> })
result = DivideIntegers.call(a: 4, b: 0)
result.failure? == true
result.type == :division
result.data == { exception: #<ZeroDivisionError: divided by 0> }
```